group-coordinator/src/main/java/org/apache/kafka/coordinator/group/generic/GenericGroup.java

```diff
@@ -833,13 +835,15 @@ public void validateOffsetCommit(
             if (generationId != this.generationId) {
                 throw Errors.ILLEGAL_GENERATION.exception();
             }
-        } else if (!isInState(EMPTY)) {
+        } else if (!isTransactional && !isInState(EMPTY)) {
             // If the request does not contain the member id and the generation id (version 0),
             // offset commits are only accepted when the group is empty.
+            // This does not apply to transactional offset commits, since the older versions
+            // of this protocol do not require member id and generation id.
             throw Errors.UNKNOWN_MEMBER_ID.exception();
         }
 
-        if (isInState(COMPLETING_REBALANCE)) {
+        if (!isTransactional && isInState(COMPLETING_REBALANCE)) {
```

Comment:
Does the comment above also apply to this case with respect to transactional offset commits?

Reply:
No. It is not applied but not for the same reason. For transactional offset commits, the group state is not checked but I have to admit that I don't know why. Let me update the comment anyway.